axlearn/common/update_transformation.py

```diff
@@ -263,3 +271,77 @@ def mask_tree(tree: dict, *, keep: dict, mask_value: Any) -> dict:
         tree,
         is_leaf=lambda x: x is None,
     )
+
+
+class OverrideInplaceUpdateTransformation(WrappedPartitionedGradientTransformation):
+    """An update transformation that provides rules to override inplace updates.
+
+    This update transformation moves gradients that match rules in `delta_updates` to
+    `inplace_updates`, then applies `PartionedGradientTransformation.update`. Also, optimizer
+    states won't be created for parameters that match these rules.
```

Comment:
Another way to look at this is that we are applying different optimizers to different params. Specifically, we apply a special optimizer for some params where the optimzier simply takes the "gradient" as the new param value.

If this makes sense, we can introduce the concept of a composite optimizer containing sub-optimizers, similar to https://github.com/apple/axlearn/blob/fde078db3a4733cfd4c3a67b83276329f2fb7716/axlearn/common/learner.py#L379-L388 and it can be configured accordingly.

Would this be a more general solution?

Reply:
Unfortunately, this cannot be implemented in learner as a regular optimizer, as all optimizer updates are delta updates, meaning that the output of the optimizer will be added to the original value. Therefore, the implementation of the "replace" logic  must be done through changing "inplace_updates".

Reply:
Using that method is not very reliable when there's inf.